pkg/controller/localvolume/add.go

```diff
@@ -0,0 +1,129 @@
+
+	localVolumeFinalizer = "storage.openshift.com/local-volume-protection"
+)
+
+// ReconcileLocalVolume reconciles a LocalVolume object
+type ReconcileLocalVolume struct {
+	// This client, initialized using mgr.Client() above, is a split client
+	// that reads objects from the cache and writes to the apiserver
+	client            client.Client
+	secClient         secv1client.SecurityV1Interface
+	scheme            *runtime.Scheme
+	apiClient         apiUpdater
+	lvMap             *common.StorageClassOwnerMap
+	controllerVersion string
+
+	localStorageNameSpace string
+	localDiskLocation     string
+	provisonerConfigName  string
+	diskMakerConfigName   string
+}
+
+// Add creates a LocalVolume Controller and adds it to the Manager. The Manager will set fields on the Controller
+// and Start it when the Manager is Started.
+func Add(mgr manager.Manager) error {
+
+	r := &ReconcileLocalVolume{
+		localDiskLocation: localDiskLocation,
+		client:            mgr.GetClient(),
+		apiClient:         newAPIUpdater(mgr),
+		secClient:         secv1client.NewForConfigOrDie(mgr.GetConfig()),
+		scheme:            mgr.GetScheme(),
+		lvMap:             &common.StorageClassOwnerMap{},
+	}
+
+	// create a new controller
+	c, err := controller.New("localvolume-controller", mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+	err = c.Watch(&source.Kind{Type: &localv1.LocalVolume{}}, &handler.EnqueueRequestForObject{})
+	if err != nil {
+		return err
+	}
+
+	err = c.Watch(&source.Kind{Type: &appsv1.DaemonSet{}}, &handler.EnqueueRequestForOwner{OwnerType: &localv1.LocalVolume{}})
+	if err != nil {
+		return err
+	}
+
+	//  watch for storageclass, enqueue owner
+	err = c.Watch(&source.Kind{Type: &corev1.PersistentVolume{}}, &handler.EnqueueRequestsFromMapFunc{
+		ToRequests: handler.ToRequestsFunc(func(obj handler.MapObject) []reconcile.Request {
+			pv, ok := obj.Object.(*corev1.PersistentVolume)
+			if !ok {
+				return []reconcile.Request{}
+			}
+
+			names := r.lvMap.GetStorageClassOwners(pv.Spec.StorageClassName)
+			reqs := make([]reconcile.Request, 0)
+			for _, name := range names {
+				reqs = append(reqs, reconcile.Request{NamespacedName: name})
+			}
+			return reqs
+		}),
+	})
+	if err != nil {
+		return err
+	}
+
+	// Watch for changes to secondary resource PersistentVolume and enqueue the LocalVolume
+	err = c.Watch(&source.Kind{Type: &corev1.PersistentVolume{}}, &handler.EnqueueRequestsFromMapFunc{
+		ToRequests: handler.ToRequestsFunc(func(obj handler.MapObject) []reconcile.Request {
+
+			pv, ok := obj.Object.(*corev1.PersistentVolume)
+			if !ok {
+				return []reconcile.Request{}
+			}
+
+			// get owner
+			ownerName, found := pv.Labels[common.PVOwnerNameLabel]
+			if !found {
+				return []reconcile.Request{}
+			}
+			ownerNamespace, found := pv.Labels[common.PVOwnerNamespaceLabel]
+			if !found {
+				return []reconcile.Request{}
+			}
+
+			// skip LocalVolumeSet owned PVs
```

Comment:
LocalVolumeSet -> LocalVolume

Reply:
This is to skip LocalVolumeSet owned PVs in the LocalVolume controller